scio-core/src/main/scala/com/spotify/scio/coders/AvroBytesUtil.scala

```diff
@@ -43,7 +43,7 @@ private[scio] object AvroBytesUtil {
     s
   }
 
-  def encode[T](coder: BCoder[T], obj: T): GenericRecord = {
+  def encode[T](coder: BCoder[T], obj: T, schema: ASchema = schema): GenericRecord = {
```

Comment:
@xBATx There's no need to pass the `schema` here and also we don't want to allow overriding.